velox/functions/prestosql/aggregates/ApproxPercentileAggregate.cpp

```diff
@@ -473,43 +473,52 @@ class ApproxPercentileAggregate : public exec::Aggregate {
       const SelectivityVector& rows,
       const BaseVector& vec) {
     DecodedVector decoded(vec, rows);
-    VELOX_USER_CHECK(
```

Comment:
This check is removed because of two reasons:

1. This check does not seems right, "Percentile argument must be constant for all input rows" does not force the input vector to be constant, right? it can be a FlatVector which contains only one distinct value.
2. The similar check is also done in `checkSetPercentile(bool,const double*,vector_size_t,std::vector<bool>)`

Reply:
Percentiles must be coming from a constant expression, that's why this check is here.  Removing this check could make some correctness issue slip though without any visible error.

Reply:
@Yuhta Can you elaborate more? a ConstantVector of value `a` seems semantically equals to a FlatVector whose values are all the same value `a`?

Reply:
Yes but you would need to go through them one by one to make sure they are really all the same

Reply:
Yeah, from performance point of view, ConstantVector is definitely the best choice for `percentile`, so we want to enforce this vector type? (Just want to confirm).

Reply:
Another perspective: The AggregationOperator(which calculates approx_percentile) might receive input from any upstream operator, we can't be sure the upstream operator produce FlatVector or ConstantVector?